File: Talleres/LaboRobot_1/WandererStrategy.java
package aspitia.mattei;

import robocode.JuniorRobot;

public class WandererStrategy implements BattleStrategy {

    private int hits = 0;
    private boolean sweepRight = true;

    @Override
    public void run(JuniorRobot robot) {
        while (true) {
            robot.turnAheadRight(30, 30);
            robot.ahead(20);
            robot.turnAheadLeft(30, 30);
            robot.turnGunRight(360);
        }
    }

    @Override
    public void onScannedRobot(JuniorRobot robot) {
        robot.turnGunTo(robot.scannedAngle);
        robot.fire(1);
    }

    @Override
    public void onHitByBullet(JuniorRobot robot) {
        robot.turnGunTo(robot.hitByBulletAngle);
        robot.fire(1);
        hits++;
        if (hits > 5) {
            hits = 0;
            robot.turnBackLeft(100, 90 - robot.hitByBulletBearing);
        }
    }

    @Override
    public void onHitWall(JuniorRobot robot) {
        robot.turnBackRight(50, 90);
    }

    @Override
    public void onHitRobot(JuniorRobot robot) {
        robot.turnGunTo(robot.hitRobotBearing);
        if (robot.energy > 1) robot.fire(3);
        robot.back(50);
    }
}

File: Talleres/LaboRobotKotlin_1/WandererStrategy.kt
package aspitia.mattei.kotlin

import robocode.JuniorRobot

class WandererStrategy : BattleStrategy {
    private var hits = 0

    override fun run(robot: JuniorRobot) {
        while (true) {
            robot.turnAheadRight(30, 30)
            robot.ahead(20)
            robot.turnAheadLeft(30, 30)
            robot.turnGunRight(360)
        }
    }

    override fun onScannedRobot(robot: JuniorRobot) {
        robot.turnGunTo(robot.scannedAngle)
        robot.fire(1.0)
    }

    override fun onHitByBullet(robot: JuniorRobot) {
        robot.turnGunTo(robot.hitByBulletAngle)
        robot.fire(1.0)
        hits++
        if (hits > 5) {
            hits = 0
            robot.turnBackLeft(100, 90 - robot.hitByBulletBearing)
        }
    }

    override fun onHitWall(robot: JuniorRobot) {
        robot.turnBackRight(50, 90)
    }

    override fun onHitRobot(robot: JuniorRobot) {
        robot.turnGunTo(robot.hitRobotBearing)
        if (robot.energy > 1) robot.fire(3.0)
        robot.back(50)
    }
}